
Automatically raise the core file size limit if CoreDumpDirectory
is configured.

--- httpd-2.3.8/server/core.c~	2010-09-01 13:15:44.000000000 +0300
+++ httpd-2.3.8/server/core.c	2010-09-01 13:16:25.533795195 +0300
@@ -3646,6 +3646,25 @@
 
     set_banner(pconf);
     ap_setup_make_content_type(pconf);
+
+#ifdef RLIMIT_CORE
+    if (ap_coredumpdir_configured) {
+        struct rlimit lim;
+
+        if (getrlimit(RLIMIT_CORE, &lim) == 0 && lim.rlim_cur == 0) {
+            lim.rlim_cur = lim.rlim_max;
+            if (setrlimit(RLIMIT_CORE, &lim) == 0) {
+                ap_log_error(APLOG_MARK, APLOG_NOTICE, 0, NULL,
+                             "core dump file size limit raised to %lu bytes",
+                             lim.rlim_cur);
+            } else {
+                ap_log_error(APLOG_MARK, APLOG_NOTICE, errno, NULL,
+                             "core dump file size is zero, setrlimit failed");
+            }
+        }
+    }
+#endif
+
     ap_setup_auth_internal(ptemp);
     if (!sys_privileges) {
         ap_log_error(APLOG_MARK, APLOG_CRIT, 0, NULL,
